awscli/data/cloudtrail/2013-11-01/completions-1.json

```diff
@@ -4,24 +4,228 @@
     "Trail": {
       "operation": "DescribeTrails",
       "resourceIdentifier": {
-        "TrailARN": "trailList[].TrailARN"
+        "Name": "trailList[].Name",
+        "S3BucketName": "trailList[].S3BucketName",
+        "S3KeyPrefix": "trailList[].S3KeyPrefix",
+        "SnsTopicName": "trailList[].SnsTopicName",
+        "IncludeGlobalServiceEvents": "trailList[].IncludeGlobalServiceEvents",
+        "IsMultiRegionTrail": "trailList[].IsMultiRegionTrail",
+        "CloudWatchLogsLogGroupArn": "trailList[].CloudWatchLogsLogGroupArn",
+        "CloudWatchLogsRoleArn": "trailList[].CloudWatchLogsRoleArn",
+        "KmsKeyId": "trailList[].KmsKeyId"
+      }
+    }
+  },
+  "operations": {
+    "CreateTrail": {
+      "Name": {
+        "completions": [
+          {
+            "parameters": {},
+            "resourceName": "Trail",
+            "resourceIdentifier": "Name"
+          }
+        ]
+      },
+      "S3BucketName": {
+        "completions": [
+          {
+            "parameters": {},
+            "resourceName": "Trail",
```

Comment:
This does not look correct as well. I would assume this would end up being a Bucket resource.

Reply:
Conceptually yes, however we don't support cross resource references so it would have to be something you can get from the cloudtrail API.  In this case, it's a little weird as you arguably shouldn't be offering auto-completion of a `Create` operation using describe/list calls for the existing service.  That would only make sense if you were trying to create a resource that's similar to one of your existing resources.  I'm considering just having an excludes list for certain operation prefixes for now (e.g `Create`).